infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLogFiles.java

```diff
@@ -298,15 +299,17 @@ public void append(long address, LogData entry) {
             if (!fh.getKnownAddresses().contains(address)) {
                 fh.getKnownAddresses().add(address);
                 writeRecord(fh, address, entry);
-
+                log.info("Disk_write[{}]: Written to disk.", address);
             } else {
-                throw new OverwriteException();
+                res = 1;     // Throw OverwriteException outside of this try/catch block
             }
-            log.info("Disk_write[{}]: Written to disk.", address);
         } catch (Exception e) {
```

Comment:
The only checked exception being thrown in this block is IOException, so catch IOException instead and throw OverwriteException when needed.